Game/Scripts/HeavyFinisherAttack.cpp

```diff
@@ -0,0 +1,315 @@
+	countEnemyHits += 1.0f;
+}
+
+void HeavyFinisherAttack::PerformHeavyFinisher(ComponentTransform* target, ComponentTransform* attackOwner)
+{
+	this->target = target;
+	this->attackOwner = attackOwner;
+
+	attackHasTarget = true;
+
+	mesh->Enable();
+	rigidBody->Disable();
+
+	owner->SetParent(loadedScene->GetRoot());
+
+	isReturningToOwner = false;
+	isActivated = true;
+}
+
+void HeavyFinisherAttack::PerformEmptyHeavyFinisher(ComponentTransform* attackOwner)
+{
+	this->attackOwner = attackOwner;
+
+	emptyAttackTargetPos = attackOwner->GetGlobalPosition() + attackOwner->GetGlobalForward().Normalized() * defaultThrowDistance;
+
+	attackHasTarget = false;
+
+	mesh->Enable();
+	rigidBody->Enable();
+
+	owner->SetParent(loadedScene->GetRoot());
+
+	std::vector<ComponentRigidBody*> collisions;
+	physics->GetCollisions(rigidBody, collisions, "Enemy"); //Get enemies inside the collider
+
+	for (ComponentRigidBody* enemy : collisions)
+	{
+		enemiesInTheArea.push_back(enemy->GetOwner()->GetComponent<ComponentTransform>());
+	}
+
+	isReturningToOwner = false;
+	isActivated = true;
+}
+
+void HeavyFinisherAttack::SeekNextEnemy()
+{
+	target = nullptr;
+
+	rigidBody->Enable();
+	rigidBody->UpdateRigidBody();
+
+	std::vector<ComponentRigidBody*> collisions;
+	physics->GetCollisions(rigidBody, collisions, "Enemy"); //Get enemies inside the collider
+
+	float3 currentPos = transform->GetGlobalPosition();
+
+	for (ComponentRigidBody* enemy : collisions)
+	{
+		ComponentTransform* enemyTransform = enemy->GetOwner()->GetComponent<ComponentTransform>();
+
+		if (!enemy->GetOwner()->GetComponent<HealthSystem>()->EntityIsAlive() //If enemy is dead
+			|| (std::find(enemiesAlreadyHit.begin(), enemiesAlreadyHit.end(), enemyTransform)
+				!= enemiesAlreadyHit.end())) //If enemy already hit
+		{
+			continue; //next enemy
+		}
+
+		if (target == nullptr) //First enemy found
+		{
+			target = enemyTransform;
+		}
+		else if (currentPos.Distance(enemyTransform->GetGlobalPosition())
+			< currentPos.Distance(target->GetGlobalPosition())) //Is a closer enemy than the selected
+		{
+			target = enemyTransform;
+		}
+	}
+
+	rigidBody->Disable();
+
+	if (target == nullptr) //No enemies to hit found
+	{
+		target = attackOwner;
+		isReturningToOwner = true;
+		if (!returnToPlayer)
+		{
+			ResetValues();
+		}
+	}
+}
+
+void HeavyFinisherAttack::ResetValues()
+{
+	transform->SetLocalPosition(attackOwner->GetGlobalPosition());
+	transform->UpdateTransformMatrices();
+	owner->SetParent(attackOwner->GetOwner());
+	target = nullptr;
+	attackOwner = nullptr;
+
+	isActivated = false;
+	isReturningToOwner = false;
+	attackHasTarget = false;
+
+	enemiesAlreadyHit.clear();
+	enemiesInTheArea.clear();
+
+	countEnemyHits = 0.0f;
+
+	vfx->Stop();
+	mesh->Disable();
+}
+
+void HeavyFinisherAttack::OnCollisionEnter(ComponentRigidBody* other)
+{
+	if (attackHasTarget)
+	{
+		return;
+	}
+
+	if (other->GetOwner()->GetTag() == "Enemy" && other->GetOwner()->IsEnabled())
```

Comment:
Based on recent pull requests, looks like the preferred way to make this check is with `CompareTag`. Not sure if there are other instances where you could change it.
```suggestion
	if (other->GetOwner()->CompareTag("Enemy") && other->GetOwner()->IsEnabled())
```